lib/data/dal.js

```diff
@@ -0,0 +1,86 @@
+var util = require('util');
+var Q = require('q');
+var regexp = require('node-regexp');
+
+var customError = require('../util/customError.js');
+
+/**
+ * Attempts to access configuration file data for the shared extension
+ *   and returns it if it exists.
+ * @param {string} confFilePath - Path and filename to the configuration file.
+ * @param {string} name - The name of the sharedExtension to access.
+ * @return {Object} - Shared extension object.
```

Comment:
You can actually describe the object being returned using something like the following:

```JavaScript
/**
 * Documentation...
 * @returns {Object} person - Person object.
 * @returns {String} person.firstName - The person's first name.
 * @returns {String} person.lastName - The person's last name.
*/
```